### PR TITLE
Re-enable leak detection parameters.

### PR DESCRIPTION
**Pull request by crummel**

### CODE REVIEW

**Review by dseefeld — APPROVED** [review]

File: eng/SourceBuild.Tarball.targets [hunk 1]
```diff
@@ -320,12 +320,12 @@
     <MSBuild
       Projects="$(LeakDetectionProjectFile)"
       Targets="Restore"
-      Properties="RestoreSources=$(SourceBuiltPackagesPath)%3B$(TarballRootDir)packages/prebuilt%3bhttps://api.nuget.org/v3/index.json"/>
+      Properties="RestoreSources=$(SourceBuiltPackagesPath)%3B$(TarballRootDir)packages/prebuilt%3bhttps://api.nuget.org/v3/index.json;DotNetPackageVersionPropsPath=$(PackageVersionPropsPath)"/>
```
Comment (dagood) [inline eng/SourceBuild.Tarball.targets:323]: A bit more readable?

```
      Properties="
        RestoreSources=$(SourceBuiltPackagesPath)%3B$(TarballRootDir)packages/prebuilt%3bhttps://api.nuget.org/v3/index.json;
        DotNetPackageVersionPropsPath=$(PackageVersionPropsPath)"/>

```

Comment (crummel) [reply eng/SourceBuild.Tarball.targets:323]: Thanks, changed as suggested.

File: eng/SourceBuild.Tarball.targets [hunk 2]
```diff
@@ -320,12 +320,14 @@
     <MSBuild
       Projects="$(LeakDetectionProjectFile)"
       Targets="Restore"
-      Properties="RestoreSources=$(SourceBuiltPackagesPath)%3B$(TarballRootDir)packages/prebuilt%3bhttps://api.nuget.org/v3/index.json"/>
+      Properties="RestoreSources=$(SourceBuiltPackagesPath)%3B$(TarballRootDir)packages/prebuilt%3bhttps://api.nuget.org/v3/index.json
+                  DotNetPackageVersionPropsPath=$(PackageVersionPropsPath)"/>
```
Comment (dagood) [inline eng/SourceBuild.Tarball.targets:324]: Don't these still need the `;`? Also... not quite my suggestion (I prefer not using alignment) but that's less important 😛:

```
      Properties="
        RestoreSources=$(SourceBuiltPackagesPath)%3B$(TarballRootDir)packages/prebuilt%3bhttps://api.nuget.org/v3/index.json;
        DotNetPackageVersionPropsPath=$(PackageVersionPropsPath)"/>

```

**Review by dagood — APPROVED** [review]

**Comment by dagood** [comment]
I forget, @crummel did you do a positive test with the infra brought back up that intentionally uses prebuilts (e.g. building without building some repos) triggers this? Were you planning to do some manual testing that's tracked somewhere?

**Comment by crummel** [comment]
Sure, filed https://github.com/dotnet/source-build/issues/1921 and I'll update that after my next manual run.

**Pull request merged**